### PR TITLE
Update tests and methods

### PR DESCRIPTION
**Pull request by NicolasBonduel**

### CODE REVIEW

File: src/collection/resources/company.js [hunk 1]
```diff
@@ -12,14 +12,6 @@ class Company extends Component {
       type: 'get', action: 'properties', urlParams, ResConstructor: null,
     }, done);
   }
-
-  get({ urlParams, json = false } = {}, done) {
```
Comment (tigo0) [inline src/collection/resources/company.js]: we don't need this anymore?

Comment (NicolasBonduel) [reply src/collection/resources/company.js]: Not for the core-sdk, users shouldn't get the whole company.

File: src/collection/resources/payments.js [hunk 2]
```diff
@@ -5,13 +5,12 @@ const {
   getFromId,
   getAll,
   get,
-  search,
   set,
 } = require('./_methods')(Payment);
 
 class Payments extends Components {
   constructor(request) {
-    super([create, getFromId, getAll, get, search, set]);
```
Comment (lazreg87) [inline src/collection/resources/payments.js]: You sure about removing the search?

File: test/architecture/architecture.json [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        ]
+      }
+    },
+    "resource": {
+      "roles": {
+        "admin": [
+          "search"
+        ],
+        "user": [
+          "create",
```
Comment (lazreg87) [inline test/architecture/architecture.json]: if we decide to keep search, add it here

File: test/architecture/index.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    assert(object[fn] !== undefined);
+    // Object functions should be accessible from the resource equivalent object.
+    assert(fn === 'refresh' || res[fn] !== undefined);
+  });
+  resourceFunctions.forEach((fn) => {
+    assert(res[fn] !== undefined);
+  });
+};
+
+describe.only('Architectural tests', () => {
```
Comment (lazreg87) [inline test/architecture/index.js]: remove .only

File: test/architecture/index.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  it('Should check architecture file correspong for AiModel', (done) => {
```
Comment (lazreg87) [inline test/architecture/index.js]: several tests are broken, need to be fixed. Mostly due to changes in the available methods for resources

**Pull request merged**